### ISSUE TITLE
[FR] Configure tracking UI layout

### ISSUE THREAD

**Issue by eliasmistler** [mlflow/mlflow#1839]
## Description / Motivation
1. We use a lot of long parameter, tag and metric names, as well as values. By default, they seem are capped at ~13-14 characters each by the CSS rule `style="display: inline-block; max-width: 120px;"`. This makes the overview page very difficult to read. 

2. Additionally, the columns `Date`, `User`, `Run Name`, `Source` and `Version` take up roughly half of the table width, while they only have one line each. We use lots of params, so the space could be used better.

3. Thirdly, the list of Tags defaults to showing 3 items and a link to "show 5 more". That seems inconsistent with the params and metrics column. Also, the latter two columns show unnecessary spacing, while the Tags column does not.

## Proposed Changes
1. Either make the `max-width` per item configurable for the UI *or*, possibly better, allow styling with additional style sheets. That would allow a fix like:

```css
span.run-table-container.underline-on-hover.metric-param-sort-toggle {
    max-width: 200px !important;
}
```

2. Possible solutions:
  - allow collapsing the meta columns into one column
  - give the user control over which columns to display

3. Apply the styling and more/less logic of the "Tags" columns to the "Parameters" and "Metrics". Ideally, add a configuration or UI control to switch off the "more/less? logic

**Comment by gioa** [comment]
Thank you for the feature request, agree space could be used better. 

Also here attached a mockup for users controlling which columns to display:  
![image](https://user-images.githubusercontent.com/1960721/65068720-fd744a00-d93d-11e9-9edb-e6af24ce34de.png)

cc @Zangr

**Comment by rom1504** [comment]
I propose to handle this in a new component for the runs, I detailed what I implemented and proposed changes there https://github.com/mlflow/mlflow/issues/2084

**Comment by Zangr** [comment]
Resolved by #2251
